### PR TITLE
feat: view local changes and refactor push and pull, deploy and retrieve e2e files

### PR DESCRIPTION
**Pull request by CristiCanizales**
Implement E2E tests for view changes commands

This PR also refactors pushAndPull.e2e.ts and deplloyAndRetrieve.e2e.ts files

[W-14866436](https://gus.lightning.force.com/a07EE00001ie4FbYAI)

Test run: https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/9127010762/job/25096411392 ✅

### CODE REVIEW

**Review by peternhale — CHANGES_REQUESTED** [review]
@CristiCanizales I really appreciate all of the refactoring you are doing to reduce duplicate code and spending time to create reusable validation functions in these tests!

@diyer

File: test/testSetup.ts [hunk 1]
```diff
@@ -466,7 +511,7 @@ export class TestSetup {
     }
   }
 
-  private async verifyProjectCreated() {
+  private async verifyProjectCreated(projectName: string) {
```
[the remark this answers is not in the document]
Comment (CristiCanizales) [reply test/testSetup.ts:468]: I still need this after my changes

**Comment by CristiCanizales** [comment]
New test run: https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/9130425241/job/25107309730 ✅

**Review by peternhale — APPROVED** [review]

File: test/specs/deployAndRetrieve.e2e.ts [hunk 2]
```diff
@@ -522,4 +335,49 @@ describe('Deploy and Retrieve', async () => {
...
+
+    // Verify Output tab
+    const outputPanelText = await utilities.attemptToFindOutputPanelText(
+      'Salesforce CLI',
+      `Starting SFDX: ${operation} Source ${fromTo}`,
+      10
+    );
+    utilities.log(
+      `${operation} time ${type} - 3: ` + (await utilities.getOperationTime(outputPanelText!))
+    );
```
Comment (daphne-sfdc) [inline test/specs/deployAndRetrieve.e2e.ts:376]: Is the `- 3` necessary in this `utilities.log()` statement?  It seems like it's being used as a count of number of times the command was run in the E2E test, which loses its meaning if it's always 3.

**Pull request merged**